quickwit/quickwit-jaeger/src/lib.rs

```diff
@@ -294,8 +297,10 @@ impl JaegerService {
                 query.push_str(" OR ");
             }
             query.push_str("trace_id:");
-            query.push_str(trace_id);
+            query.push_str(std::str::from_utf8(trace_id).expect("Base64 should be valid UTF-8."));
```

Comment:
I don't see the justification for trace_id to be valid utf-8.
Is the `trace_id` something that is passed in a request?

Maybe we could replace the type def by a `struct Base64TraceId([u8; 24])`;
and check for the str validity  in the constructors.